src/SixLabors.Fonts/TextLayout.cs

```diff
@@ -554,11 +554,18 @@ private static void SubstituteBidiMirrors(FontMetrics fontMetrics, GlyphSubstitu
                     CodePoint codePoint = codePointEnumerator.Current;
                     GlyphMetrics glyph = metrics[0];
                     float glyphAdvance = isHorizontal ? glyph.AdvanceWidth : glyph.AdvanceHeight;
+                    if (CodePoint.IsVariationSelector(codePoint))
+                    {
+                        codePointIndex++;
+                        graphemeCodePointIndex++;
+                        continue;
+                    }
+
                     if (CodePoint.IsTabulation(codePoint))
                     {
                         glyphAdvance *= options.TabWidth;
                     }
-                    else if (CodePoint.IsZeroWidthJoiner(codePoint) || CodePoint.IsZeroWidthJoiner(codePoint))
```

Comment:
I think you found a type there. The second condition should be `CodePoint.IsZeroWidthNonJoiner(codePoint)` as I recall.

Reply:
Updated! 

(I'm wondering it we should completely avoid them and not generate glyphs for them too)

Reply:
As I recall they affect bidi reordering so must be kept.